code/_onclick/hud/screen_objects.dm

```diff
@@ -445,7 +445,7 @@
 
 	if (user.max_stamina == -1 || user.stamina == user.max_stamina)
 		if (user.stamina_bar)
-			QDEL_NULL(user.stamina_bar)
+			user.stamina_bar.endProgress()
```

Comment:
This seems to leave a reference that would lead to an harddel, since user.stamina_bar will keep a reference to the progressbar object, and endProgress() queue itself to be qdeleted; The reference should be nulled afterwards, if the QDEL_NULL macro is not used